src/main/java/com/shopify/minishopify/repository/CategoryRepository.java

```diff
@@ -2,6 +2,11 @@
 
 import com.shopify.minishopify.model.Category;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
 
 public interface CategoryRepository extends JpaRepository<Category, Integer> {
+    @Query("SELECT c FROM Category c WHERE c.name LIKE %?1%")
```

Comment:
Can you add a comment to this code to explain what `%?1%` does? 

Is it finding all categories that has `c.name` in the string?
e.g., If `c.name` is `category1`, would it match **randomtextcategory1randomtext`**?

Reply:
the SQL find all entries that have the provided argument in its name field. % - wildcard similar to *, ?1 - 1st parameter

Reply:
Correct me if wrong so that would mean if I had a category named `McDonalds`, this query would be able to match:
1. `FakeMcDonalds`
2. `McDonalds`

Assuming this is true, how would you guarantee on this line:
https://github.com/mpfan/Mini-Shopify/pull/61/commits/427cbd8a3ac8e38e515aee2796a1c0e97b6ef120#diff-621bafc5ec794da125e28ba18f7f4ee48bc6a80bb93bca84f268269f9c04b848R48

that the first one you retrieve would be the category the category that the user wishes to add to a shop?

Reply:
74e45a7 changed the way this was handled. This issue is no longer pertinent.